lib/soft_deletion/core.rb

```diff
@@ -62,6 +68,12 @@ def deleted?
 
     def mark_as_deleted
       self.deleted_at ||= Time.now
+      if self.class.soft_deletion_update_timestamp
+        timestamp_variable_name = "@#{self.class.soft_deletion_update_timestamp}".to_sym
+        new_timestamp = [self.deleted_at, instance_variable_get(timestamp_variable_name)].compact.max
+
+        instance_variable_set(timestamp_variable_name, new_timestamp)
```

Comment:
`send(self.class.soft_deletion_update_timestamp)` should also work ?
same for `send("#{self.class.soft_deletion_update_timestamp}=")`
possibly safer since AR might not treat it as instance var under the hood

Reply:
I don't really like using `send`, but it's fine.